integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkScalaContainerTest.java

```diff
@@ -189,4 +196,127 @@ void testScalaUnionRddToParquet() {
+        new GenericContainer<>(DockerImageName.parse(SPARK_DOCKER_IMAGE))
+            .dependsOn(kafkaContainer)
+            .waitingFor(Wait.forLogMessage(SPARK_DOCKER_CONTAINER_WAIT_MESSAGE, 1))
+            .withNetwork(network)
+            .withStartupTimeout(Duration.ofMinutes(2));
+
+    List<String> command = new ArrayList<>();
+    command.add("./bin/spark-submit");
+    command.add("--master");
+    command.add("local[*]");
+    command.add("--class");
+    command.add(className);
+    command.add("--packages");
+    command.add(System.getProperty("kafka.package.version"));
+
+    addSparkConfig(command, "spark.driver.extraJavaOptions=-Dderby.system.home=/tmp/derby");
+    addSparkConfig(command, "spark.extraListeners=" + OpenLineageSparkListener.class.getName());
+    addSparkConfig(command, "spark.jars.ivy=/tmp/.ivy2/");
+    addSparkConfig(command, "spark.openlineage.debugFacet=enabled");
+    addSparkConfig(
+        command, "spark.openlineage.facets.disabled=[schema;spark_unknown;spark.logicalPlan]");
+    addSparkConfig(command, "spark.openlineage.transport.type=file");
+    addSparkConfig(command, "spark.openlineage.transport.location=/tmp/events.log");
+    addSparkConfig(command, "spark.sql.shuffle.partitions=1");
+    addSparkConfig(command, "spark.sql.warehouse.dir=/tmp/warehouse");
+    addSparkConfig(command, "spark.ui.enabled=false");
+    command.add(CONTAINER_FIXTURES_JAR_PATH.toString());
+
+    // mount the additional Jars
+    mountPath(spark, HOST_SCALA_FIXTURES_JAR_PATH, CONTAINER_FIXTURES_JAR_PATH);
+    mountFiles(spark, HOST_LIB_DIR, CONTAINER_SPARK_JARS_DIR);
+    mountFiles(spark, HOST_ADDITIONAL_JARS_DIR, CONTAINER_SPARK_JARS_DIR);
+    mountFiles(spark, HOST_ADDITIONAL_CONF_DIR, CONTAINER_SPARK_CONF_DIR);
+
+    final String commandStr = String.join(" ", command);
+    log.info("Command is {}", commandStr);
+
+    spark.withCommand(commandStr);
+
+    spark.start();
+
+    Awaitility.await().atMost(Duration.ofSeconds(60)).until(() -> !spark.isRunning());
+
+    kafkaInitContainer.stop();
+    kafkaContainer.stop();
+    zookeeperContainer.stop();
+
+    File temporaryFile = File.createTempFile("events", ".log");
+
+    spark.copyFileFromContainer("/tmp/events.log", temporaryFile.getPath());
+
+    temporaryFile.deleteOnExit();
+
+    List<RunEvent> events =
+        Files.readAllLines(temporaryFile.toPath()).stream()
+            .map(OpenLineageClientUtils::runEventFromJson)
+            .collect(Collectors.toList());
+
+    List<RunEvent> nonEmptyInputEvents =
+        events.stream().filter(e -> !e.getInputs().isEmpty()).collect(Collectors.toList());
+
+    assertEquals(3, nonEmptyInputEvents.size());
+
+    nonEmptyInputEvents.forEach(
+        event -> {
+          assertEquals(1, event.getInputs().size());
+          assertEquals("input-topic", event.getInputs().get(0).getName());
+          assertEquals("kafka://kafka.broker.zero:9092", event.getInputs().get(0).getNamespace());
```

Comment:
What happens if the are multiple brokers in the cluster? Does the namespace become randomly assigned?

Reply:
My thinking was that this gets solved by dataset namespace resolvers: https://openlineage.io/docs/client/java/configuration#dataset-namespace-resolver

Reply:
Oh. I forgot about that feature.

Reply:
Perhaps it makes sense to include it within one of the integration tests. Would be good as a reference I believe.

Reply:
I created a follow-up issue for that #2864